### ISSUE TITLE
reproducing contact matrices from the original publication

### ISSUE THREAD

**Issue by WastlM** [sbfnk/socialmixr#1]
Thanks for establishing this R package, which greatly simplifies the use of contact matrices in future research projects. I had already used social contact matrices from the German subset of the POLYMOD survey in earlier analyses. Now I am trying to extract the same contact matrices with **socialmixr**, but get different numbers.

My versions of the contact matrices for Germany are available in the R package [**hhh4contacts**](https://CRAN.R-project.org/package=hhh4contacts). For instance, the following matrix is identical to the one given in Mossong et al (2008, Supporting Information, Table S5 (8.2)):
```r
hhh4contacts::contactmatrix_mossong
```
```
           contact
participant 00-04 05-09 10-14 15-19 20-24 25-29 30-34 35-39 40-44 45-49 50-54 55-59 60-64 65-69  70+
      00-04  1.90  0.81  0.34  0.42  0.56  0.62  1.20  0.87  0.49  0.40  0.38  0.26  0.27  0.19 0.38
      05-09  0.36  2.25  0.61  0.42  0.24  0.43  0.66  0.77  0.72  0.38  0.23  0.26  0.22  0.18 0.63
      10-14  0.09  0.35  3.58  0.93  0.25  0.25  0.34  0.65  0.97  0.44  0.37  0.15  0.19  0.19 0.24
      15-19  0.04  0.18  0.69  4.19  1.29  0.36  0.22  0.60  0.69  0.68  0.39  0.26  0.12  0.15 0.14
      20-24  0.08  0.03  0.12  0.90  2.59  1.14  0.45  0.45  0.47  0.63  0.41  0.26  0.11  0.08 0.19
      25-29  0.27  0.17  0.16  0.28  1.43  1.79  1.09  0.77  0.60  0.77  0.59  0.59  0.49  0.23 0.28
      30-34  0.46  0.26  0.14  0.19  0.44  0.98  1.49  1.18  0.98  0.32  0.67  0.35  0.42  0.19 0.21
      35-39  0.20  0.49  0.62  0.46  0.22  0.30  0.74  1.43  1.03  0.59  0.36  0.28  0.36  0.30 0.22
      40-44  0.20  0.23  0.67  1.11  0.57  0.38  0.57  0.95  1.23  0.93  0.47  0.22  0.32  0.20 0.40
      45-49  0.09  0.09  0.35  1.21  0.45  0.57  0.59  0.70  1.05  1.23  0.82  0.41  0.28  0.15 0.26
      50-54  0.19  0.20  0.16  0.27  0.47  0.60  0.54  0.58  0.71  0.99  1.18  0.79  0.42  0.18 0.28
      55-59  0.07  0.11  0.07  0.26  0.33  0.45  0.61  0.57  0.59  0.62  0.79  1.12  0.54  0.30 0.34
      60-64  0.10  0.14  0.07  0.11  0.19  0.32  0.31  0.36  0.49  0.44  0.40  0.75  0.85  0.52 0.37
      65-69  0.03  0.11  0.18  0.15  0.15  0.19  0.32  0.42  0.62  0.38  0.45  0.35  0.80  0.91 1.30
      70+    0.07  0.27  0.34  0.49  0.30  0.24  0.41  0.50  0.52  0.37  0.38  0.31  0.43  0.44 0.76
attr(,"agedistri")
 00-04  05-09  10-14  15-19  20-24  25-29  30-34  35-39  40-44  45-49  50-54  55-59  60-64  65-69    70+ 
0.0687 0.0710 0.0849 0.0934 0.0903 0.0579 0.0440 0.0533 0.0672 0.0726 0.0695 0.0587 0.0649 0.0571 0.0463 
```

Using the POLYMOD data originally made available at https://www.researchgate.net/publication/232701632_POLYMOD_contact_survey_for_researchers, I could reproduce this table if I replace all missing values for participant age group and contact age group in the contacts dataset by "70+" (which appears strange).

The contact numbers returned by **socialmixr**`::contact_matrix()` are quiet different (while the age distribution of the participants is in agreement):
```r
library("socialmixr")
data(polymod)
CM <- contact_matrix(polymod, countries = "Germany", age.limits = seq(0,70,by=5))
round(CM$matrix, 2)
```
```
         contact.age.group
age.group [0,5) [5,10) [10,15) [15,20) [20,25) [25,30) [30,35) [35,40) [40,45) [45,50) [50,55) [55,60) [60,65) [65,70)  70+
  [0,5)    1.37   0.89    0.35    0.47    0.69    0.70    1.13    0.82    0.52    0.44    0.44    0.36    0.33    0.20 0.39
  [5,10)   0.41   2.46    0.76    0.39    0.16    0.46    0.70    0.71    0.82    0.34    0.29    0.32    0.24    0.17 0.15
  [10,15)  0.09   0.33    3.35    1.10    0.31    0.25    0.24    0.67    1.08    0.42    0.30    0.15    0.19    0.23 0.29
  [15,20)  0.04   0.20    0.81    4.40    1.08    0.31    0.19    0.55    0.63    0.71    0.42    0.25    0.11    0.16 0.14
  [20,25)  0.09   0.03    0.14    1.08    2.90    1.00    0.38    0.39    0.50    0.55    0.37    0.17    0.10    0.05 0.18
  [25,30)  0.31   0.20    0.16    0.36    1.41    1.96    1.12    0.57    0.48    0.71    0.60    0.67    0.43    0.24 0.29
  [30,35)  0.53   0.32    0.23    0.21    0.42    1.07    1.46    0.95    0.72    0.21    0.77    0.37    0.51    0.26 0.26
  [35,40)  0.23   0.68    0.80    0.57    0.13    0.29    0.71    1.49    0.90    0.29    0.28    0.32    0.29    0.39 0.26
  [40,45)  0.25   0.28    0.82    1.44    0.62    0.28    0.46    0.94    1.10    0.76    0.43    0.14    0.37    0.17 0.41
  [45,50)  0.14   0.10    0.45    1.61    0.41    0.60    0.36    0.55    1.07    1.14    0.79    0.35    0.21    0.18 0.29
  [50,55)  0.19   0.22    0.16    0.23    0.48    0.58    0.47    0.60    0.61    0.98    1.30    0.77    0.50    0.20 0.27
  [55,60)  0.08   0.16    0.07    0.25    0.29    0.45    0.54    0.54    0.42    0.59    0.84    1.26    0.53    0.25 0.49
  [60,65)  0.13   0.18    0.07    0.12    0.19    0.32    0.30    0.40    0.38    0.33    0.38    0.56    0.89    0.65 0.50
  [65,70)  0.04   0.12    0.26    0.20    0.16    0.24    0.35    0.38    0.76    0.32    0.36    0.32    0.96    0.97 0.88
  70+      0.12   0.18    0.03    0.13    0.07    0.13    0.33    0.38    0.33    0.08    0.30    0.23    0.48    0.70 1.13
```
```r
round(CM$participants$proportion, 4)
```
```
0.0687 0.0710 0.0849 0.0934 0.0903 0.0579 0.0440 0.0533 0.0672 0.0726 0.0695 0.0587 0.0649 0.0571 0.0463
```

Changing the handling of `missing.contact.age` does not yield the first matrix either.
I suspect that the differences are not solely caused by the crude replacement of missing age values by "70+" in the first matrix. The *amount of missing values* for the age of the contact person actually differs between the ResearchGate version of the contacts dataset and the version provided by **socialmixr**. According to
```r
contactsD <- merge(polymod$contacts,
                   subset(polymod$participants, country == "Germany", c(part_id, part_age)),
                   by = "part_id", all = FALSE)
```
there are `nrow(contactsD)` = 10659 contact persons of participants from Germany, of which `sum(is.na(contactsD$cnt_age))` = 2818 have no age information in "cnt_age" (which is the variable used by the `contact_matrix` function). In the corresponding contacts dataset from ResearchGate (which yields the same number of contact persons), there are only 107 missing values in the "cnt_age_mean" variable.

**socialmixr** presumably gives a better estimate of the contact matrix than the previous one. Could you elaborate on why this is and how "cnt_age" was obtained (maybe from "cnt_age_exact", "cnt_age_est_min", and "cnt_age_est_max") as opposed to "cnt_age_mean" in the older version?

**Comment by sbfnk** [comment]
Thanks for reporting this issue which is, hopefully, fixed in a version I'm about to upload (for installation with `devtools::install_github`). 

1. The `cnt_age` column in the `polymod` dataset is a derived column (from `..._exact`, `..._min` and `..._max`). It has now been removed from the `polymod` dataset. How exactly the column is derived can now be set with the `estimated.contact.age` argument in `contact_matrix`.
2. There were some differences between the POLYMOD data in Zenodo and on ResearchGate. The data in the package are now based on the ResearchGate data which appear more complete.

I now get almost the same numbers as the ones you report:
```
> CM <- contact_matrix(polymod, countries = "Germany", age.limits = seq(0,70,by=5), estimated.contact.age="mean", missing.contact.age="keep", missing.participant.age="keep")
> round(CM$matrix, 2)
         contact.age.group
age.group [0,5) [5,10) [10,15) [15,20) [20,25) [25,30) [30,35) [35,40) [40,45) [45,50) [50,55) [55,60) [60,65) [65,70)  70+ <NA>
  [0,5)    1.90   0.81    0.34    0.42    0.56    0.62    1.20    0.87    0.49    0.40    0.38    0.26    0.27    0.19 0.34 0.04
  [5,10)   0.36   2.25    0.61    0.42    0.24    0.43    0.66    0.77    0.72    0.38    0.23    0.26    0.22    0.18 0.15 0.48
  [10,15)  0.09   0.35    3.58    0.93    0.25    0.25    0.34    0.65    0.97    0.44    0.37    0.15    0.19    0.19 0.24 0.00
  [15,20)  0.04   0.18    0.69    4.19    1.29    0.36    0.22    0.60    0.69    0.68    0.39    0.26    0.12    0.15 0.13 0.01
  [20,25)  0.08   0.03    0.12    0.90    2.59    1.14    0.45    0.45    0.47    0.63    0.41    0.26    0.11    0.08 0.19 0.00
  [25,30)  0.27   0.17    0.16    0.28    1.43    1.79    1.09    0.77    0.60    0.77    0.59    0.59    0.49    0.23 0.24 0.04
  [30,35)  0.46   0.26    0.14    0.19    0.44    0.98    1.49    1.18    0.98    0.32    0.67    0.35    0.42    0.19 0.21 0.00
  [35,40)  0.20   0.49    0.62    0.46    0.22    0.30    0.74    1.43    1.03    0.59    0.36    0.28    0.36    0.30 0.22 0.00
  [40,45)  0.20   0.23    0.67    1.11    0.57    0.38    0.57    0.95    1.23    0.93    0.47    0.22    0.32    0.20 0.39 0.01
  [45,50)  0.09   0.09    0.35    1.21    0.45    0.57    0.59    0.70    1.05    1.23    0.82    0.41    0.28    0.15 0.24 0.01
  [50,55)  0.19   0.20    0.16    0.27    0.47    0.60    0.54    0.58    0.71    0.99    1.18    0.79    0.42    0.18 0.26 0.02
  [55,60)  0.07   0.11    0.07    0.26    0.33    0.45    0.61    0.57    0.59    0.62    0.79    1.12    0.54    0.30 0.32 0.03
  [60,65)  0.10   0.14    0.07    0.11    0.19    0.32    0.31    0.36    0.49    0.44    0.40    0.75    0.85    0.52 0.36 0.01
  [65,70)  0.03   0.11    0.18    0.15    0.15    0.19    0.32    0.42    0.62    0.38    0.45    0.35    0.80    0.91 0.76 0.54
  70+      0.08   0.17    0.05    0.13    0.07    0.18    0.32    0.38    0.27    0.22    0.32    0.22    0.52    0.60 1.07 0.07
  <NA>     0.04   0.41    0.72    0.96    0.61    0.30    0.52    0.65    0.85    0.57    0.46    0.43    0.33    0.24 0.20 0.09
```

The only remaining difference is in the 70+ age group, which I presume is because of the treatment of missing ages as you suggest. Feel free to re-open this issue if you think anything else is still not right.

**Comment by WastlM** [comment]
Does the newer "Version 1.1" on Zenodo (https://doi.org/10.5281/zenodo.1157934), uploaded on January 23, 2018, now correspond to the ResearchGate data and to the data in the package?

Using the current development version of **socialmixr**,
```r
cite(polymod)
```
points to the previous "Version 1.1" at https://doi.org/10.5281/zenodo.1059920.

**Comment by sbfnk** [comment]
Thanks for pointing this out - both development and (just released) version 0.1.2 now use Version 1.1 of the data set on Zenodo. I would expect this to match the ResearchGate data (but haven't done a detailed check).
